okhttp/src/main/java/com/backblaze/b2/client/okHttpClient/B2OkHttpClientImpl.java

```diff
@@ -0,0 +1,327 @@
+
+    @Override
+    public void getContent(String url,
+                           B2Headers headersOrNull,
+                           B2ContentSink handler) throws B2Exception {
+        Request.Builder builder = new Request.Builder()
+                .get()
+                .url(url);
+        Headers headers = makeOkHeaders(headersOrNull);
+        if( headers != null ){
+            builder.headers(headers);
+        }
+        Request request = builder.build();
+
+        try (Response response = okHttpClient.newCall(request).execute()) {
+            if (!response.isSuccessful()) {
+                throw extractExceptionFromErrorResponse(response, response.body().string());
+            } else {
+                if( response.body() != null ){
+                    Headers responseHeaders = response.headers();
+                    B2Headers b2headers = makeB2Headers(responseHeaders);
+                    handler.readContent(b2headers, response.body().byteStream());
+                }
+            }
+        } catch (IOException e) {
+            throw translateToB2Exception(e, url);
+        }
+    }
+
+    /**
+     * HEADSs to a web service that returns content, and returns the headers.
+     *
+     * @param url the url to head to
+     * @param headersOrNull the headers, if any.
+     * @return the headers of the response.
+     * @throws B2Exception if there's any trouble
+     */
+    @Override
+    public B2Headers head(String url, B2Headers headersOrNull)
+            throws B2Exception {
+        Request.Builder builder = new Request.Builder()
+                .url(url)
+                .head();
+        Headers reqHeaders = makeOkHeaders(headersOrNull);
+        if( reqHeaders != null ){
+            builder.headers(reqHeaders);
+        }
+        Request request = builder.build();
+        try (Response response = okHttpClient.newCall(request).execute()) {
+            if (!response.isSuccessful()) {
+                throw extractExceptionFromErrorResponse(response, response.body().string());
+            } else {
+                if( response.code() == 200){
+                    Headers responseHeaders = response.headers();
+                    return makeB2Headers(responseHeaders);
+                } else {
+                    throw B2Exception.create(null, response.code(), null, "");
+                }
+            }
+        } catch (IOException e) {
+            throw translateToB2Exception(e, url);
+        }
+    }
+
+    @Override
+    public void close() {
+        // Closing response body streams is handled elsewhere by OkHttp.
+    }
+
+    private String postJsonAndReturnString(String url,
+                                           B2Headers headersOrNull,
+                                           Object request) throws B2Exception {
+        try {
+            B2Json bzJson = B2Json.get();
+            String requestJson = bzJson.toJson(request);
+            byte[] requestBytes = getUtf8Bytes(requestJson);
```

Comment:
hmmm....hidden limitation that we can't handle more than 2GB-ish of json in our requests.  that's the same thing exists in the Apache HttpClient-based implementation.  i think that's fine -- B2 request json structures are relatively small.